src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.cs

```diff
@@ -618,13 +623,14 @@ private void InitializeEventDetectionThread()
                 {
                     IsBackground = true
                 };
+                _eventThreadCancellationTokenSource = new CancellationTokenSource();
                 _eventDetectionThread.Start();
             }
         }
 
         private void DetectEvents()
         {
-            while (_pinsToDetectEventsCount > 0)
+            while (_pinsToDetectEventsCount > 0 && _eventThreadCancellationTokenSource != null)
```

Comment:
```suggestion
            if (_eventThreadCancellationTokenSource == null)
            {
                throw new InvalidOperationException("Cannot start to detect events when CancellationTokenSource is null.");
            }

            while (_pinsToDetectEventsCount > 0)
```

The other two places where you added the null check are just cancelling and disposing the token source (and not setting it to null which I think in this case is correct) and TBH I'm fine with the way you have it where if the condition evaluates to false (meaning the cancellation token source is null) then it is fine to just silently continue, since all those methods are trying to do is to cancel the token and then wait for the thread to exit.

That said, for this use I don't like the fact that we are adding the condition to the while, because a) this should never be true as in theory the CTS should be initialized by `InitializeEventDetectionThread` method before calling this, and b) if that is ever not the case (meaning if the CTS is null on this thread) I much rather consider that as a failed state and we should throw an exception instead of just failing silently, as nobody is setting the CTS to null and nobody should call it while CTS hasn't been initialized.

Reply:
If you agree with my suggestion, I'll go ahead and commit the suggestion and merge the PR as I think apart from that comment this is ready to go.